src/app/api/v1/retrofunding/rounds/[roundId]/impactMetrics/[impactMetricId]/comments/route.ts

```diff
@@ -8,22 +11,50 @@ export async function GET(
   request: NextRequest,
   route: { params: { roundId: string; impactMetricId: string } }
 ) {
-  const { roundId, impactMetricId } = route.params;
-  const comments = fetchImpactMetricComments(roundId, impactMetricId);
-  return new Response(JSON.stringify(comments), { status: 200 });
+  const authResponse = await authenticateApiUser(request);
+
+  if (!authResponse.authenticated) {
+    return new Response(authResponse.reason, { status: 401 });
+  }
+
+  return await traceWithUserId(authResponse.userId as string, async () => {
```

Comment:
room for thought: we could probably refactor this replicated code into a wrapper